codersdk/richparameters_test.go

```diff
@@ -304,3 +304,89 @@ func TestRichParameterValidation(t *testing.T) {
 		}
 	})
 }
+
+func TestParameterResolver_ValidateResolve_Ephemeral_OverridePrevious(t *testing.T) {
+	t.Parallel()
+	uut := codersdk.ParameterResolver{
+		Rich: []codersdk.WorkspaceBuildParameter{{Name: "n", Value: "5"}},
+	}
+	p := codersdk.TemplateVersionParameter{
+		Name:      "n",
+		Type:      "number",
+		Mutable:   true,
+		Required:  true,
+		Ephemeral: true,
+	}
+	v, err := uut.ValidateResolve(p, &codersdk.WorkspaceBuildParameter{
+		Name:  "n",
+		Value: "6",
+	})
+	require.NoError(t, err)
+	require.Equal(t, "6", v)
+}
+
+func TestParameterResolver_ValidateResolve_Ephemeral_FirstTime(t *testing.T) {
+	t.Parallel()
+	uut := codersdk.ParameterResolver{}
+	p := codersdk.TemplateVersionParameter{
+		Name:      "n",
+		Type:      "number",
+		Mutable:   true,
+		Required:  true,
+		Ephemeral: true,
+	}
+	v, err := uut.ValidateResolve(p, &codersdk.WorkspaceBuildParameter{
+		Name:  "n",
+		Value: "6",
+	})
+	require.NoError(t, err)
+	require.Equal(t, "6", v)
+}
+
+func TestParameterResolver_ValidateResolve_Ephemeral_UseDefault(t *testing.T) {
+	t.Parallel()
+	uut := codersdk.ParameterResolver{}
+	p := codersdk.TemplateVersionParameter{
+		Name:         "n",
+		Type:         "number",
+		Mutable:      true,
+		DefaultValue: "5",
+		Ephemeral:    true,
+	}
+	v, err := uut.ValidateResolve(p, nil)
+	require.NoError(t, err)
+	require.Equal(t, "5", v)
+}
+
+func TestParameterResolver_ValidateResolve_Ephemeral_UseEmptyDefault(t *testing.T) {
+	t.Parallel()
+	uut := codersdk.ParameterResolver{}
+	p := codersdk.TemplateVersionParameter{
+		Name:         "n",
+		Type:         "number",
+		Mutable:      true,
+		DefaultValue: "",
+		Ephemeral:    true,
+	}
+	v, err := uut.ValidateResolve(p, nil)
+	require.NoError(t, err)
+	require.Equal(t, "", v)
+}
+
+func TestParameterResolver_ValidateResolve_Ephemeral_RequiredButMissing(t *testing.T) {
```

Comment:
What about a case where required = true, but there's also a default value set? Is this possible and should we test it?

Reply:
It is not possible as `required` is a computed property, the opposite of [optional](https://github.com/coder/terraform-provider-coder/blob/main/provider/parameter.go#L105C5-L111), converted [here](https://github.com/coder/coder/blob/main/provisioner/terraform/resources.go#L500).